### DISCUSSION TITLE
Clarify license?

### DISCUSSION THREAD

**Discussion by niloc132** [webfx-project/webfx#20] [Q&A]
According to the repo license information and the LICENSE file, this project is Apache v2, but it looks like there are a few files which are explicitly licensed as being owned by Oracle, perhaps copied directly from another incompatibly-licensed project?

A few examples:

https://github.com/webfx-project/webfx/blob/129538899ecb42518542d1438c47afc83ba309d9/webfx-kit/webfx-kit-javafxbase-emul/src/main/java/com/sun/javafx/binding/BidirectionalBinding.java#L1-L4

https://github.com/webfx-project/webfx/blob/129538899ecb42518542d1438c47afc83ba309d9/webfx-kit/webfx-kit-javafxbase-emul/src/main/java/com/sun/javafx/collections/ObservableMapWrapper.java#L1-L4

https://github.com/webfx-project/webfx/blob/129538899ecb42518542d1438c47afc83ba309d9/webfx-kit/webfx-kit-javafxbase-emul/src/main/java/javafx/event/Event.java#L1-L4

Can you confirm that these are clean-room implementations that somehow have the wrong copyright header, or which packages/modules are safe to use under the Apache v2 license? 

Thanks!

**Comment by salmonb** [comment]
Thanks for your question.

We publish our work under Apache 2 license. This license applies to our work only (i.e. the source code we are the author of). So the clarification you're asking comes by understanding what exactly is that source code we are the author of, and that is therefore subject to our license.

The WebFX project has several repositories, mainly the WebFX Platform, WebFX Kit, WebFX Stack, and WebFX Extras. 
We are the author of the WebFX Platform, WebFX Stack, and WebFX Extras (these repositories are also published under Apache 2 license).
Regarding the WebFX Kit (which is the concern of your question), it is divided into 2 parts: the higher layer - which provides a patched version of OpenJFX compatible with GWT, and the lower layer - which is a scene graph -> DOM mapper.
We are the author of the lower layer, and the author of the patch required for the higher layer, but of course not of OpenJFX itself. The patch refers only to the modification of the OpenJFX source code that is required to make it GWT compatible. 

Currently the patching process is manual, because historically this is how this work started in 2015, while investigating the feasibility of compiling OpenJFX with GWT.
We plan to improve this patching process and automate it before the General Availability of WebFX (as mentioned in our [roadmap](https://github.com/webfx-project/webfx/blob/main/.github/ROADMAP.md)).
Once automated, only the patch will be stored in the WebFX repository, and the compilation will consist of taking the OpenJFX source code from the Maven repository, apply that patch in a hot manner (without the need to store the result in our repository) and then give that result as input to the GWT compiler.

But because we haven't reached that stage yet, for now we technically need to store the result of that manual patching process in our repository. This is just a technical requirement, and it doesn't mean that we own this source code. To make evidence of that, we intentionally kept the statements you are referring to (these statements are coming from the OpenJFX code).

Hopefully the confusion this may create will disappear when we will automate this process and keep only the patch stored in our repository. 

In summary, the source code that is subject to our Apache 2 license is the WebFX Platform, WebFX Stack, WebFX Extras, and regarding the WebFX Kit, the scene graph -> DOM mapper and the patch. Our Apache 2 license doesn't apply to the remaining source code of OpenJFX that is temporary stored in our repository for the technical reasons explained above.
